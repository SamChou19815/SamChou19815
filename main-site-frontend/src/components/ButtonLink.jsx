// @flow strict

import type { Node } from 'react';
import React from 'react';
// $FlowFixMe
import Button from '@material-ui/core/Button';
// $FlowFixMe
import Link from '@material-ui/core/Link';

type Props = {|
  +href: string;
  +children: Node;
  +color: string;
  +openInNewTab: boolean;
  +onClick?: () => void;
  +className?: string;
|};

const ButtonLink = ({ href, children, color, openInNewTab, onClick, className }: Props): Node => (
  <Button color={color} className={className} onClick={onClick}>
    <Link
      color={color}
      href={href === '' ? undefined : href}
      target={openInNewTab ? '_blank' : undefined}
      rel="noopener noreferrer"
    >
      {children}
    </Link>
  </Button>
);

ButtonLink.defaultProps = {
  color: 'primary',
  openInNewTab: false,
};

export default ButtonLink;
