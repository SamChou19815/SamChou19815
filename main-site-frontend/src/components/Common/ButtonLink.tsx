import React, { ReactElement, ReactNode } from 'react';
import Button from '@material-ui/core/Button';
import Link from '@material-ui/core/Link';

type Props = {
  readonly href: string;
  readonly children: ReactNode;
  readonly color: 'primary' | 'secondary' | 'inherit';
  readonly openInNewTab: boolean;
  readonly onClick?: () => void;
  readonly className?: string;
};

const ButtonLink = ({
  href,
  children,
  color,
  openInNewTab,
  onClick,
  className,
}: Props): ReactElement => (
  <Button color={color} className={className} onClick={onClick}>
    <Link
      color={color}
      href={href === '' ? undefined : href}
      target={openInNewTab ? '_blank' : undefined}
      rel="noopener noreferrer"
    >
      {children}
    </Link>
  </Button>
);

ButtonLink.defaultProps = {
  color: 'primary',
  openInNewTab: false,
};

export default ButtonLink;
