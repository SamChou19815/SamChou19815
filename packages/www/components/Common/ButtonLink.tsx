import React, { ReactElement } from 'react';

import classnames from 'classnames';

type Props = {
  readonly href: string;
  readonly className?: string;
  readonly children: string;
};

const ButtonLink = ({ href, children, className }: Props): ReactElement => (
  <a className={classnames('button', 'button--link', className)} href={href}>
    {children.toLocaleUpperCase()}
  </a>
);

export default ButtonLink;
