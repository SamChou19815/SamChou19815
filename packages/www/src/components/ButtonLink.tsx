import type { ReactNode } from 'react';

const ButtonLinkBaseCSS = [
  'button bg-transparent border-0 cursor-pointer px-6 py-1.5 font-bold text-sm text-center',
  'hover:bg-blue-500 hover:bg-opacity-10',
].join(' ');

type Props = { readonly href: string; readonly className?: string; readonly children: ReactNode };

export default function ButtonLink({ href, children, className }: Props): JSX.Element {
  return (
    <a
      className={className == null ? ButtonLinkBaseCSS : `${ButtonLinkBaseCSS} ${className}`}
      href={href}
    >
      {typeof children === 'string' ? children.toLocaleUpperCase() : children}
    </a>
  );
}
