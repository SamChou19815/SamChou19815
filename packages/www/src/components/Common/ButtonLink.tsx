import clsx from 'clsx';
import React, { ReactElement, ReactNode } from 'react';

type Props = {
  readonly href: string;
  readonly className?: string;
  readonly children: ReactNode;
};

const ButtonLink = ({ href, children, className }: Props): ReactElement => (
  <a className={clsx('button', 'button--link', className)} href={href}>
    {typeof children === 'string' ? children.toLocaleUpperCase() : children}
  </a>
);

export default ButtonLink;
