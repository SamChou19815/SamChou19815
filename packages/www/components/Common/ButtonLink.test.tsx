import React from 'react';

import renderer from 'react-test-renderer';

import ButtonLink from './ButtonLink';

it('Simple ButtonLink matches snapshot.', () => {
  const tree = renderer.create(<ButtonLink href="haha">I am a button</ButtonLink>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Simple ButtonLink with class matches snapshot.', () => {
  const tree = renderer
    .create(
      <ButtonLink href="haha" className="clazz">
        I am a button
      </ButtonLink>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
