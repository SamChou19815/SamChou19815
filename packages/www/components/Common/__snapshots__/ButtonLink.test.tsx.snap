// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple ButtonLink matches snapshot. 1`] = `
<a
  className="button button--link"
  href="haha"
>
  I AM A BUTTON
</a>
`;

exports[`Simple ButtonLink with class matches snapshot. 1`] = `
<a
  className="button button--link clazz"
  href="haha"
>
  I AM A BUTTON
</a>
`;
